### PR TITLE
Case Distribution | Update legacy and direct review proportion levers based on the Board's request

### PR DESCRIPTION
**Pull request by alisan16**
Context on the requested change in this Slack thread: https://dsva.slack.com/archives/CJL810329/p1614280894080100

* [New settings for levers](https://docs.google.com/presentation/d/10S5k-5oU1hPordx-zmrR_rKNYNbyC7KjyBNMhDbZ9Sk/edit#slide=id.gc0c1b17b92_2_0) (✅ confirmed in writing from the Board on 3/2/2021)
* [Refresher on automatic case distribution](https://docs.google.com/presentation/d/10S5k-5oU1hPordx-zmrR_rKNYNbyC7KjyBNMhDbZ9Sk/edit#slide=id.g54fd333abc_0_70)

### Description
Update the docket proportions calculations in step 4 of the algorithm to match the Board's priority of having 90% of the distributions be legacy cases.
> 4. Distribute non-priority cases that are general population based on docket proportions

The docket proportions will be:
* Legacy: Set proportion of 90%
* Direct review: Maximum proportion of 7%
* Hearings and evidence submission: Remaining proportion after the first two are calculated based on number of ready cases in those dockets
The sum of all proportions should add up to 1.

This PR also removes the interpolated minimum direct review proportion and pacesetting direct review proportion calculations as requested by the Board. Since we are more than a year out post AMA day (Feb 19, 2019), the Board would like to only work direct reviews that are due, not based on the rate that they arrive.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] `MINIMUM_LEGACY_PROPORTION` = 0.9 and `MAXIMUM_DIRECT_REVIEW_PROPORTION` = 0.07 based on the Board's requirements

### Testing Plan
1. Run tests in spec/models/docket_coordinator_spec.rb

### CODE REVIEW

**Comment by alisan16** [comment]
Working through test failures in [distribution_spec.rb](https://github.com/department-of-veterans-affairs/caseflow/blob/master/spec/models/distribution_spec.rb) now 

Open question: this won’t come up practically for a long time period (months, a year?), but **what should happen when there aren't enough legacy cases to meet the 90% minimum?** There legacy case inventory is over 100,000 right now so we aren’t close to hitting this. In that case, the Chairman indicated that we should be distributing due direct reviews and then the hearings/evidence submission AMA cases proportionally. Current implementation does not handle that case. Should that be included in this change?

The Chairman also requested for more direct control over the levers (a way for the Board to change them as they wish), which may translate to us making more manual changes in the coming months until that functionality is prioritized and built out in a future PI.

**Comment by yoomlam** [comment]
> Working through test failures in [distribution_spec.rb](https://github.com/department-of-veterans-affairs/caseflow/blob/master/spec/models/distribution_spec.rb) now
> 
> Open question: this won’t come up practically for a long time period (months, a year?), but **what should happen when there aren't enough legacy cases to meet the 90% minimum?** The legacy case inventory is over 100,000 right now so we aren’t close to hitting this. In that case, the Chairman indicated that we should be distributing due direct reviews and then the hearings/evidence submission AMA cases proportionally. Current implementation does not handle that case. Should that be included in this change?
>
> The Chairman also requested for more direct control over the levers (a way for the Board to change them as they wish), which may translate to us making more manual changes in the coming months until that functionality is prioritized and built out in a future PI.

Since we're going to be making changes in the coming months, I'd say don't include it, but definitely put this note in future ticket(s) to make sure we make algorithm updates to address this.

**Comment by n1zyy** [comment]
> Open question: this won’t come up practically for a long time period (months, a year?), but **what should happen when there aren't enough legacy cases to meet the 90% minimum?** The legacy case inventory is over 100,000 right now so we aren’t close to hitting this. In that case, the Chairman indicated that we should be distributing due direct reviews and then the hearings/evidence submission AMA cases proportionally. Current implementation does not handle that case. Should that be included in this change?

I'd +1 what Yoom said. (I also think this will be a great problem to have!)

As a relative newcomer to this section of code, it feels somewhat Byzantine, so I like that you've ripped out some no-longer-applicable code. I'll all in favor of just tackling the next case when we start to approach that rather than trying to handle future cases now.

File: app/models/concerns/automatic_case_distribution.rb [hunk 1]
```diff
@@ -7,8 +7,6 @@ module AutomaticCaseDistribution
            :docket_proportions,
            :priority_count,
            :direct_review_due_count,
-           :pacesetting_direct_review_proportion,
```
Comment (alisan16) [inline app/models/concerns/automatic_case_distribution.rb:10]: Remove pacesetting and interpolated proportions, we are no longer using them now that we are 1 year post AMA

File: app/models/docket_coordinator.rb [hunk 2]
```diff
@@ -111,32 +109,14 @@ def direct_review_due_count
...
-    receipts_per_year = dockets[:direct_review].nonpriority_receipts_per_year
-
-    @pacesetting_direct_review_proportion = receipts_per_year.to_f / nonpriority_decisions_per_year
-  end
-
   def total_batch_size
     JudgeTeam.includes(:non_admin_users).flat_map(&:non_admin_users).size * Distribution::CASES_PER_ATTORNEY
   end
 
+  def due_direct_review_proportion
```
Comment (alisan16) [inline app/models/docket_coordinator.rb:116]: Pulled this out into a method to reference in tests

File: spec/models/docket_coordinator_spec.rb [hunk 3]
```diff
@@ -1,22 +1,18 @@
 # frozen_string_literal: true
 
 describe DocketCoordinator do
-  describe "direct review docket pace setting" do
+  describe "direct review docket steady state" do
```
Comment (alisan16) [inline spec/models/docket_coordinator_spec.rb:4]: No longer using "pace setting" proportions. We are now in the "steady state" world. See more info here - https://docs.google.com/presentation/d/10S5k-5oU1hPordx-zmrR_rKNYNbyC7KjyBNMhDbZ9Sk/edit#slide=id.g87305c4e85_0_57

File: spec/models/docket_coordinator_spec.rb [hunk 4]
```diff
@@ -1,22 +1,18 @@
...
     before do
       FeatureToggle.enable!(:test_facols)
       Timecop.freeze(Time.utc(2020, 4, 1, 12, 0, 0))
 
-      4.times do
+      10.times do
```
Comment (alisan16) [inline spec/models/docket_coordinator_spec.rb:9]: Just to make the math easier. Total batch size = # of judges x # attorneys per judge x ATTORNEY_BATCH_SIZE = 10 x 5 x 3 = 150

File: spec/models/docket_coordinator_spec.rb [hunk 5]
```diff
@@ -52,7 +47,7 @@
       end
     end
 
-    let(:nonpriority_legacy_count) { 10 }
+    let(:nonpriority_legacy_count) { 130 }
```
Comment (alisan16) [inline spec/models/docket_coordinator_spec.rb:50]: legacy cases = 130 nonpriority + 10 priority (see line 30) = 140
total batch size = 150
docket_margin_net_of_priority = total batch size - priority cases = 150 - 140

legacy proportion = 140 / 140 = 100%, above MINIMUM_LEGACY_PROPORTION OF 90%

File: spec/models/docket_coordinator_spec.rb [hunk 6]
```diff
@@ -121,17 +116,23 @@
...
+    end
+
     context "when there are due direct reviews" do
       it "uses the number of due direct reviews as a proportion of the docket margin net of priority" do
-        expect(docket_coordinator.docket_proportions).to eq(
-          legacy: 0.4,
-          direct_review: 0.2,
-          evidence_submission: 0.2,
-          hearing: 0.2
+        expect(docket_coordinator.docket_proportions).to include(
```
Comment (alisan16) [inline spec/models/docket_coordinator_spec.rb:128]: Based on due direct review number on line 74
5 / 140 = 0.0357

File: spec/models/docket_coordinator_spec.rb [hunk 7]
```diff
@@ -121,17 +116,23 @@
...
+          direct_review: docket_coordinator.due_direct_review_proportion
         )
-        expect(docket_coordinator.pacesetting_direct_review_proportion).to eq(0.1)
-        expect(docket_coordinator.interpolated_minimum_direct_review_proportion).to eq(0.067)
-        expect(docket_coordinator.target_number_of_ama_hearings(2.years)).to eq(400)
+        expect(docket_coordinator.target_number_of_ama_hearings(2.years)).to eq(64)
```
Comment (alisan16) [inline spec/models/docket_coordinator_spec.rb:131]: Method here - https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/docket_coordinator.rb#L76

```    
decisions_in_days = (time_period.to_f / 1.year) * nonpriority_decisions_per_year
(decisions_in_days * docket_proportions[:hearing]).round
```
docket proportion for hearings = (1 - 0.9 - 0.0357) / 2 = 0.03214

```
decisions_in_days = 2 * 1000
(2000 * 0.03214).round = 64

File: spec/models/docket_coordinator_spec.rb [hunk 8]
```diff
@@ -156,56 +157,55 @@
...
 
         context "unless there aren't that many cases" do
-          let(:nonpriority_legacy_count) { 3 }
+          let(:nonpriority_legacy_count) { 15 }
 
           it "uses the maximum number possible" do
             expect(docket_coordinator.docket_proportions).to include(
-              legacy: 0.05,
-              direct_review: 0.8
+              legacy: 0.1
```
Comment (alisan16) [inline spec/models/docket_coordinator_spec.rb:191]: 15 / 150 = 0.1

File: spec/models/distribution_spec.rb [hunk 9]
```diff
@@ -203,9 +208,9 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
             expect(VACOLS::CaseDocket.nonpriority_hearing_cases_for_judge_count(judge))
               .to eq total_tied_nonpriority_hearings
             subject.distribute!
-
             expect(subject.valid?).to eq(true)
-            expect(subject.statistics["legacy_hearing_backlog_count"]).to eq(BACKLOG_LIMIT)
+            # This may be less than BACKLOG_LIMIT when MINIMUM_LEGACY_PROPORTION is very large
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:212]: MINIMUM_LEGACY_PROPORTION is set to 90% with this PR. With this high of a proportion, more nonpriority hearings cases will be distributed, potentially overreaching the BACKLOG_LIMIT of 30

File: spec/models/distribution_spec.rb [hunk 10]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.valid?).to eq(true)
+          expect(subject.priority_push).to eq(false)
+          expect(subject.status).to eq("completed")
+          expect(subject.started_at).to eq(Time.zone.now) # time is frozen so appears zero time elapsed
+          expect(subject.errored_at).to be_nil
+          expect(subject.completed_at).to eq(Time.zone.now)
+          expect(subject.statistics["batch_size"]).to eq(15)
+          expect(subject.statistics["total_batch_size"]).to eq(45)
+          expect(subject.statistics["priority_count"]).to eq(legacy_priority_count + 1)
+          expect(subject.statistics["legacy_proportion"]).to eq(DocketCoordinator::MINIMUM_LEGACY_PROPORTION)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:171]: Changed from 0.4

File: spec/models/distribution_spec.rb [hunk 11]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.statistics["legacy_hearing_backlog_count"]).to be <= 3
+          expect(subject.statistics["direct_review_proportion"]).to eq(DocketCoordinator::MAXIMUM_DIRECT_REVIEW_PROPORTION)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:173]: Changed from 0.2

File: spec/models/distribution_spec.rb [hunk 12]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.statistics["evidence_submission_proportion"]).to be_within(0.01).of(other_dockets_proportion)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:174]: Changed from 0.2

File: spec/models/distribution_spec.rb [hunk 13]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.statistics["hearing_proportion"]).to be_within(0.01).of(other_dockets_proportion)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:175]: Changed from 0.2

File: spec/models/distribution_spec.rb [hunk 14]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.statistics["nonpriority_iterations"]).to be_between(1, 3)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:176]: Changed from be_between(2, 3)

File: spec/models/distribution_spec.rb [hunk 15]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.distributed_cases.count).to eq(15)
+          expect(subject.distributed_cases.first.docket).to eq("legacy")
+          expect(subject.distributed_cases.first.ready_at).to eq(2.days.ago.beginning_of_day)
+          expect(subject.distributed_cases.where(priority: true).count).to eq(5)
+          expect(subject.distributed_cases.where(genpop: true).count).to be_within(1).of(7)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:181]: Changed from

`expect(subject.distributed_cases.where(genpop: true).count).to eq(5)`

File: spec/models/distribution_spec.rb [hunk 16]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.distributed_cases.where(priority: true, genpop: false).count).to eq(2)
+          expect(subject.distributed_cases.where(priority: false, genpop_query: "not_genpop").count).to eq(0)
+          expect(subject.distributed_cases.where(priority: false, genpop_query: "any").map(&:docket_index).max).to eq(35)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:184]: Changed from
`expect(subject.distributed_cases.where(priority: false, genpop_query: "any").map(&:docket_index).max).to eq(30)`

File: spec/models/distribution_spec.rb [hunk 17]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.distributed_cases.where(priority: true,
+                                                 docket: Constants.AMA_DOCKETS.direct_review).count).to eq(1)
+          expect(subject.distributed_cases.where(docket: "legacy").count).to be >= 8
+          expect(subject.distributed_cases.where(docket: Constants.AMA_DOCKETS.direct_review).count).to be >= 2
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:188]: Changed from
`expect(subject.distributed_cases.where(docket: Constants.AMA_DOCKETS.direct_review).count).to be >= 3`

File: spec/models/distribution_spec.rb [hunk 18]
```diff
@@ -146,43 +146,48 @@ def create_legacy_case_hearing_for(appeal, board_member: judge.vacols_attorney_i
...
+          expect(subject.distributed_cases.where(docket: Constants.AMA_DOCKETS.evidence_submission).count).to eq(0)
```
Comment (alisan16) [inline spec/models/distribution_spec.rb:189]: Changed from
`expect(subject.distributed_cases.where(docket: Constants.AMA_DOCKETS.evidence_submission).count).to eq(2)`

File: spec/models/distribution_spec.rb (hunk 11 above)
Comment (yoomlam) [reply spec/models/distribution_spec.rb:173]: For others examining these changes, enable "Hide whitespace changes" to see all the diffs:
image

File: spec/models/distribution_spec.rb [hunk 19]
```diff
@@ -412,6 +417,8 @@ def create_priority_distributed_case(distribution, case_id, ready_at)
         let!(:legacy_nonpriority_cases) { [] }
         let!(:same_judge_nonpriority_hearings) { [] }
         let!(:other_judge_hearings) { [] }
+        # Proportion for hearings and evidence submission dockets
+        let(:other_dockets_proportion) { (1 - DocketCoordinator::MAXIMUM_DIRECT_REVIEW_PROPORTION) / 2 }
```
Comment (yoomlam) [inline spec/models/distribution_spec.rb:421]: Seem like this should account for legacy cases too:
```suggestion
        let(:other_dockets_proportion) { (1 - DocketCoordinator::MINIMUM_LEGACY_PROPORTION - DocketCoordinator::MAXIMUM_DIRECT_REVIEW_PROPORTION) / 2 }
```

Comment (yoomlam) [reply spec/models/distribution_spec.rb:421]: Nevermind, the test scenario is `when there are zero legacy cases eligible`.

**Review by yoomlam — APPROVED** [review]
LGTM 🚀

**Comment by alisan16** [comment]
Getting this test failure in CircleCI, but unable to reproduce locally
```
  1) PushPriorityAppealsToJudgesJob.eligible_judge_target_distributions_with_leftovers tracking distributions over time evens out over multiple calls
     Failure/Error: expect(final_counts.max - final_counts.min).to be <= 1

       expected: <= 1
            got:    4
     # ./spec/jobs/push_priority_appeals_to_judges_job_spec.rb:479:in `block (4 levels) in <top (required)>'
```

**Comment by yoomlam** [comment]
> Getting this test failure in CircleCI, but unable to reproduce locally

I got it to fail with this configuration:
```ruby
"number_judges: 9, @distribution_counts: {0=>10, 1=>9, 2=>4, 3=>5, 4=>1, 5=>1, 6=>11, 7=>2, 8=>7}"
       expected: <= 1
            got:    2
```

These other configurations succeeded:
```ruby
"number_judges: 9, @distribution_counts: {0=>1, 1=>1, 2=>3, 3=>4, 4=>8, 5=>3, 6=>5, 7=>6, 8=>3}"

"number_judges: 9, @distribution_counts: {0=>7, 1=>9, 2=>4, 3=>2, 4=>4, 5=>9, 6=>5, 7=>0, 8=>6}"

"number_judges: 8, @distribution_counts: {0=>9, 1=>10, 2=>11, 3=>1, 4=>9, 5=>4, 6=>11, 7=>8}"

"number_judges: 6, @distribution_counts: {0=>0, 1=>0, 2=>4, 3=>3, 4=>6, 5=>4}"

"number_judges: 9, @distribution_counts: {0=>4, 1=>1, 2=>1, 3=>3, 4=>10, 5=>8, 6=>3, 7=>3, 8=>8}"

"number_judges: 8, @distribution_counts: {0=>9, 1=>11, 2=>2, 3=>9, 4=>11, 5=>7, 6=>1, 7=>6}"

"number_judges: 10, @distribution_counts: {0=>4, 1=>4, 2=>11, 3=>10, 4=>8, 5=>7, 6=>10, 7=>9, 8=>11, 9=>3}"

"number_judges: 8, @distribution_counts: {0=>0, 1=>2, 2=>11, 3=>10, 4=>6, 5=>6, 6=>2, 7=>3}"

"number_judges: 9, @distribution_counts: {0=>0, 1=>0, 2=>6, 3=>10, 4=>10, 5=>0, 6=>7, 7=>1, 8=>9}"

"number_judges: 6, @distribution_counts: {0=>0, 1=>0, 2=>10, 3=>7, 4=>6, 5=>9}"

"number_judges: 10, @distribution_counts: {0=>7, 1=>9, 2=>9, 3=>9, 4=>2, 5=>6, 6=>6, 7=>0, 8=>2, 9=>11}"

"number_judges: 7, @distribution_counts: {0=>3, 1=>11, 2=>9, 3=>8, 4=>7, 5=>1, 6=>1}"

"number_judges: 9, @distribution_counts: {0=>0, 1=>3, 2=>1, 3=>9, 4=>1, 5=>2, 6=>4, 7=>6, 8=>1}"

"number_judges: 8, @distribution_counts: {0=>3, 1=>7, 2=>7, 3=>0, 4=>4, 5=>9, 6=>2, 7=>4}"

"number_judges: 5, @distribution_counts: {0=>8, 1=>11, 2=>3, 3=>9, 4=>2}"

"number_judges: 5, @distribution_counts: {0=>11, 1=>8, 2=>10, 3=>3, 4=>10}"

"number_judges: 5, @distribution_counts: {0=>2, 1=>8, 2=>2, 3=>8, 4=>5}"

"number_judges: 6, @distribution_counts: {0=>6, 1=>10, 2=>9, 3=>3, 4=>8, 5=>6}"

"number_judges: 9, @distribution_counts: {0=>4, 1=>7, 2=>9, 3=>4, 4=>1, 5=>8, 6=>11, 7=>9, 8=>11}"
```

I ran the test multiple times and printed the configuration like so:
```ruby
    20.times do context "tracking distributions over time" do
      let(:number_judges) { rand(5..10) }
      let(:priority_count) { rand(10..30) }

      before do
        # Mock cases already distributed this month
        @distribution_counts = to_judge_hash(Array.new(number_judges).map { rand(12) })
        pp "number_judges: #{number_judges}, @distribution_counts: #{@distribution_counts}"
```

**Pull request merged**